package-res/resources/web/pentaho/visual/base/model.js

```diff
@@ -45,37 +73,300 @@ define([
      * @description Creates a base `Model`.
      * @param {pentaho.visual.base.spec.IModel} modelSpec A plain object containing the model specification.
      */
-    var Model = Complex.extend({
-          meta: {
-            id: "pentaho/visual/base",
-            view: "View",
-            isAbstract: true,
-            props: [
-              {
-                name: "width",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "height",
-                type: "number",
-                isRequired: true
-              },
-              {
-                name: "isInteractive",
-                type: "boolean",
-                value: true
-              },
-              {
-                name: "data",
-                type: "object",
-                isRequired: true
+    var Model = Complex.extend(/** @lends pentaho.visual.base.Model# */{
+        constructor: function(modelSpec) {
+          this.base(modelSpec);
+        },
+
+        //region Event Flows Handling
+        /**
+         * Modifies the current selection filter based on an input filter and on a selection mode.
+         *
+         * This action is the entry point for user-driven modifications of the current selection filter,
+         * e.g. if the user clicked a bar in a bar chart,
+         * or drew a rectangle over a set of bars in a bar chart.
+         *
+         * The event ["will:select"]{@link pentaho.visual.base.events.WillSelect}
+         * is first emitted.
+         * Its event listeners can be attributed a _priority_,
+         * and can be regarded as operations in a processing pipeline that are allowed to:
+         * - cancel the event,
+         * - replace the input filter
+         * - replace the selection mode.
+         *
+         * Afterwards, [_changeSelection]{@link pentaho.visual.base.Model#_changeSelection}
```

Comment:
Previous comment: "Suggest moving "afterwards" to the end of the sentence, since [_changeSelection]{ ... } is the topic."

So like this? 
[_changeSelection]{@link pentaho.visual.base.Model#_changeSelection} is invoked to compute and update the current selection filter afterwards.

Or should I keep the comma?